bin/comment_with_dependency_diff

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -euo pipefail
```

Comment:
📚  The more we add utility scripts like those to this plugin, the more I realize we should have made a better job as providing help / documentation at what each of those do. Especially so that people not as familiar with our CI (e.g. devs that are interested in using some of those helper scripts in their pipelines themselves but have no idea what each utility does) can better understand the purpose of each.

For example, [the comment at the start of `comment_on_pr` is a good example of having some code-comment documentation on what the utility does and how it's expected to be used](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/120/files#diff-d536c5bec9be6d9bc743cc0175b25457a7417a97a3afded630db75505b91882cL3-R22)

I wish we took that habit sooner and applied it to all of our other `bin/*` utilities in this repo, making them all start with a code-comment explaining their role / context in which those helpers would be used and could help, documenting what the script does and its expected parameters and inputs (e.g. "this script should only run after `./gradlew …` has been called and generated the build artifacts", etc).
We should probably make a pass at all of them at some point.

In the meantime, wdyt about us already starting moving towards this good habit and add some intro doc comments in any new helpers we introduce, starting from this one?